packages/accounts-password/password_server.js

```diff
@@ -602,39 +560,45 @@ Accounts.sendEnrollmentEmail = function (userId, email) {
     email: email,
     when: when
   };
-  Meteor.users.update(userId, {$set: {
-    "services.password.reset": tokenRecord
-  }});
-
+  Meteor.users.update(userId, {
+    $set: {
+      "services.password.reset": tokenRecord
+    }
+  });
   // before passing to template, update user object with new token
   Meteor._ensure(user, 'services', 'password').reset = tokenRecord;
 
-  var enrollAccountUrl = Accounts.urls.enrollAccount(token);
+  var url;
+
+  if (type === 'resetPassword') {
+    url = Accounts.urls.resetPassword(token);
+  }
+
+  if (type === 'enrollAccount') {
+    url = Accounts.urls.enrollAccount(token);
+  }
 
   var options = {
     to: email,
-    from: Accounts.emailTemplates.enrollAccount.from
-      ? Accounts.emailTemplates.enrollAccount.from(user)
-      : Accounts.emailTemplates.from,
-    subject: Accounts.emailTemplates.enrollAccount.subject(user)
+    from: Accounts.emailTemplates[type].from ? Accounts.emailTemplates[type].from(user) : Accounts.emailTemplates.from,
+    subject: Accounts.emailTemplates[type].subject(user)
   };
 
-  if (typeof Accounts.emailTemplates.enrollAccount.text === 'function') {
+  if (typeof Accounts.emailTemplates[type].text === 'function') {
```

Comment:
This will throw can't access text of undefined when type isn't provided, and there's no check if it's provided.